packages/arb-token-bridge-ui/src/components/TransferPanel/TokenSearch.tsx

```diff
@@ -558,43 +553,9 @@ export function TokenSearch({
     }
 
     try {
-      // Native USDC on L2 won't have a corresponding L1 address
```

Comment:
Moved this logic to `useSelectedToken`.

`useSelectedToken` now only requires `erc20ParentAddress` and all the computation whether it's USDC or not is handled there.